src/main/java/org/reactome/release/cosmicupdate/COSMICUpdateUtil.java

```diff
@@ -0,0 +1,320 @@
+			}
+			else
+			{
+				nonEWASPrinter.printRecord(identifier, ewas.toString());
+			}
+		}
+	}
+
+	/**
+	 * Checks modifiedResidues (only FragmentReplacedModification and FragmentInsertionModification are of interest) to see if they match
+	 * refSequence. 
+	 * @param refSequence A Reference Sequence
+	 * @param modResidues The modified residues.
+	 * @return TRUE if there is a mismatch: a mismatch is when the reference sequence DBID != the modified residues's reference sequence's DBID. FALSE, otherwise.
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 */
+	private static boolean referenceSequenceMismatchesResidues(GKInstance refSequence, List<GKInstance> modResidues) throws InvalidAttributeException, Exception
+	{
+		boolean foundMismatchedRefSequence = false;
+		int i = 0;
+		while (!foundMismatchedRefSequence && i < modResidues.size())
+		{
+			GKInstance modResidue = modResidues.get(i);
+			if (modResidue.getSchemClass().getName().contains(ReactomeJavaConstants.FragmentReplacedModification)
+				|| modResidue.getSchemClass().getName().contains(ReactomeJavaConstants.FragmentInsertionModification))
+			{
+				GKInstance residueRefSequence = (GKInstance) modResidue.getAttributeValue(ReactomeJavaConstants.referenceSequence);
+				foundMismatchedRefSequence = !residueRefSequence.getDBID().equals(refSequence.getDBID());
+			}
+			i++;
+		}
+		return foundMismatchedRefSequence;
+	}
+
+	/**
+	 * Gets COSMIC identifiers from the database.
+	 * Queries the database for a ReferenceDatabase named "COSMIC" and then gets all DatabaseIdentifier objects
+	 * that refer to the COSMIC ReferenceDatabase via the referenceDatabase attribute.
+	 * This method will terminate the execution of the program if more than 1 "COSMIC" ReferenceDatabase is found.
+	 * If you plan to add more "COSMIC" ReferenceDatabase objects, this code will need to be changed to use the <em>correct</em> "COSMIC"
+	 * ReferenceDatabase.
+	 * @param adaptor
+	 * @return A Collection of DatabaseIdentifier objects.
+	 * @throws SQLException
+	 * @throws Exception
+	 * @throws InvalidAttributeException
+	 */
+	static Collection<GKInstance> getCOSMICIdentifiers(MySQLAdaptor adaptor) throws SQLException, Exception, InvalidAttributeException
+	{
+		@SuppressWarnings("unchecked")
+		Collection<GKInstance> refDBs = adaptor.fetchInstanceByAttribute(ReactomeJavaConstants.ReferenceDatabase, ReactomeJavaConstants.name, " = ", "COSMIC");
+		GKInstance cosmicRefDB = null;
+		if (refDBs.size() == 1)
+		{
+			cosmicRefDB = refDBs.stream().findFirst().get();
+		}
+		else
+		{
+			logger.error("Wrong number of COSMIC refDBs: {} ; only 1 was expected.", refDBs.size());
+			logger.error("Cannot proceeed!");
+			System.exit(1);
+		}
+		
+		@SuppressWarnings("unchecked")
+		Collection<GKInstance> cosmicObjects = adaptor.fetchInstanceByAttribute(ReactomeJavaConstants.DatabaseIdentifier, ReactomeJavaConstants.referenceDatabase, " = ", cosmicRefDB.getAttributeValue(ReactomeJavaConstants.DB_ID));
```

Comment:
I didn't realize you could do this - so you are searching for instances that have the COSMIC refDB, and rather than feeding it the refDB instance, you can just feed it the DBID? Thats neat!

Reply:
`fetchInstanceByAttribute` is a very useful function.